### PR TITLE
 ⚠️ Add OwnerReference to Kubeconfig secret

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/cluster_controller.go [hunk 1]
```diff
@@ -216,6 +217,20 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		}
 	}
 
+	// Delete the Cluster's Kubeconfig Secret, if any.
```
Comment (detiber) [inline controllers/cluster_controller.go]: If there is an OwnerReference, this shouldn't be needed.

Comment (vincepri) [reply controllers/cluster_controller.go]: We don't set one today, should we?

Comment (detiber) [reply controllers/cluster_controller.go]: We should be adding the OwnerRef when we create, and if someone is providing a pre-existing kubeconfig secret, then we should likely not assume that we can delete it for them.

Comment (vincepri) [reply controllers/cluster_controller.go]: Changed it up to add the owner ref

**Comment by ncdc** [comment]
Should we add a unit test?

**Comment by vincepri** [comment]
@ncdc I might need to follow-up with a clusterctl fix as well to remove the OwnerRef if it's a cluster one, I'll look into it later today

**Pull request merged**